skimage/transform/registration.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+from scipy.optimize import basinhopping, minimize
+from scipy import ndimage as ndi
+
+from .pyramids import pyramid_gaussian
+from ..measure import compare_mse
+
+__all__ = ['register_affine']
+
+
+def _p_to_matrix(p):
+    """
+    Converts a transformation of the first 2 rows of a 3x3 transformation matrix
+    to the full matrix
+
+    Parameters
+    ----------
+    param : array
+        Input array giving the affine transformation paramaters
+
+    Returns
+    -------
+    matrix : (3, 3) array
+        A transformation matrix used to obtain a new image
+    """
+
+    out = np.empty((3, 3))
+
+    out[0][0] = p[0]
+    out[0][1] = p[1]
+    out[0][2] = p[2]
+    out[1][0] = p[3]
+    out[1][1] = p[4]
+    out[1][2] = p[5]
+    out[2][0] = 0
+    out[2][1] = 0
+    out[2][2] = 1
+
+    return out
+
+
+def _matrix_to_p(matrix):
+    """
+    Converts a 3x3 transformation matrix into a transformation in form of first
+    2 rows unravelled
+
+    Parameters
+    ----------
+    matrix : (3, 3) array
+        A transformation matrix used to obtain a new image
+
+    Returns
+    -------
+    param : array
+        Input array giving the affine transformation of an image
+
+    """
+
+    return (matrix[0][0], matrix[0][1], matrix[0][2], matrix[1][0], matrix[1][1], matrix[1][2])
+
+
+def register_affine(reference, target, *, cost=compare_mse, nlevels=None,
+                    iter_callback=lambda img, matrix: None):
+
```

Comment:
Super important to have a docstring here - this is the function accessible to users, so it should have great documentation (including a short example).